pypesto/sample/pymc.py

```diff
@@ -216,6 +216,89 @@ def sample(self, n_samples: int, beta: float = 1.0):
 
         self.data = data
 
+    def fit(
+        self,
+        n_iterations: int,
+        method: str = 'advi',
+        random_seed: Optional[int] = None,
+        start_sigma: Optional = None,
+        inf_kwargs: Optional = None,
+        beta: float = 1.0,
+        **kwargs,
+    ):
+        """
+        Sample the problem.
+
+        Parameters
+        ----------
+        n_iterations:
+            Number of iterations.
+        method: str or :class:`Inference`
+            string name is case-insensitive in:
+            -   'advi'  for ADVI
+            -   'fullrank_advi'  for FullRankADVI
+            -   'svgd'  for Stein Variational Gradient Descent
+            -   'asvgd'  for Amortized Stein Variational Gradient Descent
+        random_seed: int
+            random seed for reproducibility
+        start_sigma: `dict[str, np.ndarray]`
+            starting standard deviation for inference, only available for method 'advi'
+        inf_kwargs: dict
+            additional kwargs passed to pymc.Inference
+        beta:
+            Inverse temperature (e.g. in parallel tempering).
+        """
+        try:
+            import pymc
+        except ImportError:
+            raise SamplerImportError("pymc")
+
+        problem = self.problem
+        log_post = PymcObjectiveOp.create_instance(problem.objective, beta)
+
+        x0 = None
+        x_names_free = problem.get_reduced_vector(problem.x_names)
+        if self.x0 is not None:
+            x0 = {
+                x_name: val
+                for x_name, val in zip(problem.x_names, self.x0)
+                if x_name in x_names_free
+            }
+
+        # create model context
+        with pymc.Model():
+            # parameter bounds as uniform prior
+            _k = [
+                pymc.Uniform(x_name, lower=lb, upper=ub)
+                for x_name, lb, ub in zip(
+                    x_names_free,
+                    problem.lb,
+                    problem.ub,
+                )
+            ]
+
+            # convert parameters to PyTensor tensor variable
+            theta = pt.as_tensor_variable(_k)
+
+            # define distribution with log-posterior as density
+            pymc.Potential("potential", log_post(theta))
+
+            # record function values
+            pymc.Deterministic("loggyposty", log_post(theta))
+
+            # perform the actual sampling
+            data = pymc.fit(
+                n=int(n_iterations),
+                method=method,
+                random_seed=random_seed,
+                start=x0,
+                start_sigma=start_sigma,
+                inf_kwargs=inf_kwargs,
+                **kwargs,
+            )
```

Comment:
As far as I could see (might have missed something), this is the only difference aside from Line 171 `trace = self.trace` to the functionality of `sample`. In this case, it might make sense to wrap this into the function either with an additional parameter or with a decorator?